conans/client/cmd/export_pkg.py

```diff
@@ -72,7 +72,7 @@ def export_pkg(app, ref, source_folder, build_folder, package_folder,
         else:
             prev = run_package_method(conanfile, package_id, hook_manager, conan_file_path, ref)
 
-    pref = PackageReference(pref.ref, pref.id, prev)
+    pref.revision = prev
```

Comment:
This is what can fail, because if you change by reference, the ``assign_prev`` doesn't have the old and the new reference to do the move (as the pref.revision also changes layout.pref.revision)